### PR TITLE
Organisms: SmallModal, MediumModal, LargeModal (Implemented as <FormModal size={"small" | "medium" | "large"}/>)

### PR DESCRIPTION
**Pull request by chamod-gamage**
Note: I've added the /adminmodal route for easy testing but I'll remove it as the last commit upon this PR's approval
Note: With these three new widths we've maxed out the number of available widths for Bootstrap Modals 
**Work Done:**

+ Further abstracted our modal component hierarchy: CommonModal --> AuthModal --> SignInModal, SignUpModal, etc.; AdminModal --> CreateRequest(''/Type/Group), etc. (future)
+ Created AdminModal which supports three sizes (props.size = "small" or "medium" or "large") according to the Notion tickets/[Figma](https://www.figma.com/file/dcSbJOfK3Jg1gCpWOd6hAK/TPC-Employee-Experience%3A-Phase-1?node-id=761%3A5675)
  + [SmallModal](https://www.notion.so/uwblueprintexecs/Organism-SmallModal-dc7a3259aaec4cc686670590f78a2a43)
image
  + [MediumModal](https://www.notion.so/uwblueprintexecs/Organism-MediumModal-6cb948bda87f4202b37005d2bdf4e28b)
image
  + [LargeModal](https://www.notion.so/uwblueprintexecs/Organism-LargeModal-88d649bf56a5480f811e94f77c1507c4)
image

List of props for AdminModal:
image

New SS:
image
image
image

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -11,12 +12,13 @@ import SampleContainer from "./components/examples/SampleContainer";
...
 import SignUpModal from "./pages/SignUpModal";
 
+
 function App(): JSX.Element {
   return (
     <span>
       <Router>
         <Switch>
-          <Route path='/signup' strict component={SignUpModal}></Route>
+          <Route path='/adminmodal' strict component={() => <AdminModal size="large" title="Create Type" handleClose={() => { console.log("HI") }} show >Create Type</AdminModal>}></Route>
```
Comment (meganniu) [inline client/src/App.tsx]: Note for other reviewers: there's a duplicate route for '/signup', so it's OK to remove '/signup' on line 19

File: client/src/App.tsx [hunk 2]
```diff
@@ -11,12 +12,13 @@ import SampleContainer from "./components/examples/SampleContainer";
 import SignInModal from "./pages/SignInModal";
 import SignUpModal from "./pages/SignUpModal";
 
+
 function App(): JSX.Element {
   return (
     <span>
       <Router>
         <Switch>
-          <Route path='/signup' strict component={SignUpModal}></Route>
```
Comment (chamod-gamage) [inline client/src/App.tsx:19]: this /signup route was a duplicate btw so i guess we'll remove it in this pr

File: client/src/components/organisms/AdminModal.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+interface Props {
+  title: string;
+  show: boolean;
+  handleClose(): void;
+  children: React.ReactNode;
+  size: "small" | "medium" | "large";
+}
+
+const AdminModal: FunctionComponent<Props> = (props: Props) => {
```
Comment (meganniu) [inline client/src/components/organisms/AdminModal.tsx]: Could we change the name of this component to `FormModal`?

File: client/src/components/organisms/AuthModal.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+interface Props {
+  title: string;
+  subtitle: string;
+  show: boolean;
+  handleClose(): void;
+  body: React.ReactNode;
+}
+
+const AuthModal: FunctionComponent<Props> = (props: Props) => {
```
Comment (meganniu) [inline client/src/components/organisms/AuthModal.tsx]: Could we change the name of this component to `LogoModal`?

File: client/src/components/organisms/Modal.tsx [hunk 5]
```diff
@@ -1,36 +1,34 @@
...
+  class: string;
+  header?: React.ReactNode;
+  customClose?: boolean;
+  size: "small" | "medium" | "large" | "default";
 }
 
+
+
 const CommonModal: FunctionComponent<Props> = (props: Props) => {
+  const modalSize: "sm" | "lg" | "xl" = props.size === "small" ? "sm" : props.size === "medium" ? "lg" : "xl"
```
Comment (meganniu) [inline client/src/components/organisms/Modal.tsx]: Could we use switch cases for this translation? LOL i think itll be more readable than nested ternary

File: client/src/components/organisms/Modal.tsx [hunk 6]
```diff
@@ -1,36 +1,34 @@
...
-        <div className="subtitle">
-          {props.subtitle}
-        </div>
-        {props.body}
-      </Modal.Body>
-      <Modal.Footer>
-      </Modal.Footer>
-    </Modal>
+    <>
+      {props.size !== "default" ?
```
Comment (meganniu) [inline client/src/components/organisms/Modal.tsx]: We could deduplicate some html by returning the following
```
<Modal show={props.show} onHide={props.handleClose} centered={true} className={props.class} size={props.size === "default" ? undefined : modalSize} dialogClassName="border-radius-12">
  <Modal.Header closeButton={!props.customClose} className={`${props.class}-header` + (props.size === "default" ? "" : `-${props.size}`)} >{props.header}{props.customClose && <i onClick={props.handleClose} className={`bi bi-x ${props.class}-header close` + (props.size === "default" ? "" : ` ${props.size}`)}></i>}</Modal.Header>
  <Modal.Body>{props.children}</Modal.Body>
  <Modal.Footer />
</Modal>
```

**Review by meganniu — CHANGES_REQUESTED** [review]
Great work Chamod!!

Styling nits:
+ "X" buttons should be black
+ Font weight of headers should be 600
+ There are some headers that include more than text. Is there some way we can accomodate this? example:
Screenshot from 2021-04-08 19-59-25

File: client/src/App.tsx [hunk 7]
```diff
@@ -11,12 +12,13 @@ import SampleContainer from "./components/examples/SampleContainer";
...
+          <Route path='/adminmodal' strict component={() => <AdminModal size="medium" title="Create Request" handleClose={() => { console.log("HI") }} show ><></></AdminModal>}></Route>
```
Comment (guy-ro) [inline client/src/App.tsx]: Remove this before merging.

File: client/src/components/organisms/AdminModal.tsx (hunk 3 above)
Comment (guy-ro) [reply client/src/components/organisms/AdminModal.tsx]: What makes this distinct from CommonModal? I don't really see anything that CommonModal can't do? The only thing different that I can see is the "hr". We can simply add an option (a boolean prop) for an "hr" in this position in CommonModal. Just making sure we really need this modal done as a separate component?

File: client/src/components/organisms/AdminModal.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  return (
+    <CommonModal customClose class="modal-admin" size={props.size} show={props.show} handleClose={props.handleClose} header={<><div className={`modal-admin-header-title-${props.size}`}>
```
Comment (guy-ro) [inline client/src/components/organisms/AdminModal.tsx]: Don't need `<>` around the div inside header

File: client/src/components/organisms/Modal.tsx (hunk 5 above)
Comment (guy-ro) [reply client/src/components/organisms/Modal.tsx]: Or instead, can we just do this through css? The size prop can be a className and the style associated with each size can be done in CommonModal.scss

**Review by chamod-gamage — COMMENTED** [review]
> Great work Chamod!!
> 
> Styling nits:
> 
> * "X" buttons should be black
> * Font weight of headers should be 600
> * There are some headers that include more than text. Is there some way we can accomodate this? example:
>   Screenshot from 2021-04-08 19-59-25

@meganniu Since every FormModal includes a left-aligned title (from what I can tell) can I include an additional prop in FormModal (typed as React.ReactNode) called extraHeader that basically takes in any JSX in cases like these?

**Review by guy-ro — APPROVED** [review]
LGTM

imo, if the sizing was done through setting the class of the outermost div in Modal.tsx to the class described in the prop "size", this would be cleaner

**Comment by chamod-gamage** [comment]
@meganniu @guy-ro Thx for your reviews :)
I've covered all of the fixes you guys brought up and all that's left is further discussion of the necessity of the component. My opinion is that maybe the work itself wasn't super necessary but now that we have the component we might as well use it since it only makes it easier down the line to plug in the Modal Body contents for the Admin Forms.

**Review by meganniu — APPROVED** [review]
Looks great :)))) Thanks @chamod-gamage

**Pull request merged**